ctk/common/src/main/kotlin/org/codice/compliance/utils/report/Report.kt

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (c) 2018 Codice Foundation
+
+Released under the GNU Lesser General Public License; see
+http://www.gnu.org/licenses/lgpl.html
+*/
+package org.codice.compliance.utils.report
+
+import org.codice.compliance.QUIET_REPORT
+import org.codice.compliance.SAMLComplianceException
+import org.fusesource.jansi.Ansi
+import org.junit.platform.engine.TestExecutionResult.Status.FAILED
+import org.junit.platform.engine.TestExecutionResult.Status.SUCCESSFUL
+
+object Report {
+
+    private val errorMessages = mutableMapOf<Sections, MutableList<SAMLComplianceException>?>()
+    private val coreHeaderSize = Sections.CORE_1.name.length
+
+    init {
+        Sections.values().forEach { errorMessages[it] = mutableListOf() }
+    }
+
+    private val reportQuietly by lazy {
+        System.getProperty(QUIET_REPORT) == "true"
+    }
+
+    fun addErrorMessage(section: Sections, error: SAMLComplianceException): Report {
+        // Error already added
+        if (errorMessages[section]?.isNotEmpty() == true &&
```

Comment:
✏️ The second test should be sufficient. I'm not sure there's enough of a performance improvement here to warrant the compound condition.

On second look, does it really matter at all if the error is added to the section multiple times? If you convert it from a mutableList to a mutableSet, at least?

Reply:
Not if it’s a set